metrics_api/test/opentelemetry/metrics/meter_test.rb

```diff
@@ -25,49 +25,259 @@
+      invalid_descriptions.each do |invalid_description|
+        _(-> { meter.create_observable_gauge('an_instrument', description: invalid_description, callback: -> {}) })
+          .must_raise(instrument_description_error, "should have raised for description=#{invalid_description.inspect}")
+      end
+    end
+
+    it 'creates and returns an instance of ObservableGauge instrument' do
+      assert(meter.create_observable_gauge('an_instrument', callback: -> {})
+        .is_a?(OpenTelemetry::Metrics::Instrument::ObservableGauge))
+    end
+
+    it 'logs a warning message when instrument with same name already exists' do
+      OpenTelemetry::TestHelpers.with_test_logger do |log_stream|
+        meter.create_observable_gauge('an_instrument', callback: -> {})
+        meter.create_observable_gauge('AN_INSTRUMENT', callback: -> {})
+        _(log_stream.string).must_match(/duplicate instrument registration occurred for instrument an_instrument/)
+      end
+    end
+  end
+
+  describe '#create_observable_up_down_counter' do
+    it 'raises an error when name does not match pattern' do
+      invalid_names.each do |invalid_name|
+        _(-> { meter.create_observable_up_down_counter(invalid_name, callback: -> {}) })
+          .must_raise(instrument_name_error, "should have raised for name=#{invalid_name.inspect}")
+      end
+    end
+
+    it 'raises an error when unit is invalid' do
+      invalid_units.each do |invalid_unit|
+        _(-> { meter.create_observable_up_down_counter('an_instrument', unit: invalid_unit, callback: -> {}) })
+          .must_raise(instrument_unit_error, "should have raised for unit=#{invalid_unit.inspect}")
+      end
+    end
+
+    it 'raises an error when description is invalid' do
+      invalid_descriptions.each do |invalid_description|
+        _(-> { meter.create_observable_up_down_counter('an_instrument', description: invalid_description, callback: -> {}) })
+          .must_raise(instrument_description_error, "should have raised for description=#{invalid_description.inspect}")
+      end
+    end
+
+    it 'creates and returns an instance of ObservableUpDownCounter instrument' do
+      assert(meter.create_observable_up_down_counter('an_instrument', callback: -> {})
+        .is_a?(OpenTelemetry::Metrics::Instrument::ObservableUpDownCounter))
+    end
+
+    it 'logs a warning message when instrument with same name already exists' do
+      OpenTelemetry::TestHelpers.with_test_logger do |log_stream|
+        meter.create_observable_up_down_counter('an_instrument', callback: -> {})
+        meter.create_observable_up_down_counter('AN_INSTRUMENT', callback: -> {})
+        _(log_stream.string).must_match(/duplicate instrument registration occurred for instrument an_instrument/)
+      end
+    end
+  end
+
+  let(:invalid_names) do
+    [
+      nil,
+      '',
+      ' ',
+      '1_meter',
+      '_meter',
+      '-meter',
+      '.meter',
+      'a' * 64,
+      'a@',
+      'a!'
+    ]
+  end
```

Comment:
I usually don't write DRY tests, but in this case I think having a singular place to define invalid names/units/descriptions helps because all of them use the same validation rules.